android/quest/src/main/java/org/smartregister/fhircore/quest/ui/patient/details/QuestPatientDetailActivity.kt

```diff
@@ -167,9 +168,55 @@ class QuestPatientDetailActivity :
   }
 
   private fun onTestResultItemClickListener(resultItem: QuestResultItem?) {
-    resultItem?.let { parser?.onResultItemClicked(resultItem, this, patientId) }
+    getResultDetailsNavigationOptions().navigationOptions.forEach {
+      handleNavigationOptions(it, resultItem, patientId)
+    }
+  }
+
+  private fun handleNavigationOptions(
+    navigationOption: NavigationOption,
+    resultItem: QuestResultItem?,
+    patientId: String
+  ) {
+    when (navigationOption.action) {
+      is QuestionnaireNavigationAction -> {
+        resultItem?.let {
+          val questionnaireResponse = resultItem.source.first
+
+          val questionnaireId = questionnaireResponse.questionnaire.split("/")[1]
```

Comment:
where is the structure of the `questionnaire` field defined?

I'm concerned about referencing a specific array item that result in an exception with unexpected input.

Reply:
The structure of questionnaire field is -> "questionnaire": "Questionnaire/3440/_history/20"

Reply:
Also adding comment to the code.

Reply:
is defined in a fhir resource? can we also add a link to the official resource that defines this?

we want the comment to show we have a guaranteed structure here and therefore it is OK to run the split and array reference.

hmm, even so, for the sake of defensive coding, I wonder if we should checking the length before referencing the index, I'm not sure if that's needed, @ellykits do you think that's necessary?

Reply:
I can see some examples on our FHIR-Server resources and one of which is here https://fhir.labs.smartregister.org/fhir/QuestionnaireResponse/54753b27-ba8c-46a7-beed-47d79743a08c#L14. Anyways, I am open to suggestions to resolve it in a more generic and safe way.

Reply:
The extra check is necessary to prevent running into `IndexOutOfBoundsExceptions`, so yes we need it. It's always a good practice to address all the known edge cases in code.

Reply:
We can show relevant error message to the user when such exception occurs.

Reply:
Easiest would be to check if the array is not empty

Reply:
I don't think that will be sufficient, since we're referencing the 2nd element, we need to check that the array has length >1

let's also add a test for the a string that fails that check

Reply:
I added it that way as suggested by @pld .

Reply:
cool, lets be sure to add tests here